doc/Type/independent-routines.pod6

```diff
@@ -1213,14 +1213,14 @@ for an example.
 
 Defined as:
 
-    multi sub shift(@container) is raw
+    multi sub shift(@collection) is raw
```

Comment:
This is simply defined as 

https://github.com/rakudo/rakudo/blob/db074a8bc49a0dbcdfc4ecd392c7eaadb0c078ed/src/core.c/array_operators.pm6#L26

As long as we're changing it, the best is to just use the original, as is the default policy.

Reply:
In your review of PR #3973 you still agreed that it was better to use more meaningful parameter names.

I don't mind changing the parameter name if you now think that `@a` is better after all, because the `@` sigil probably adds enough meaning. But as an aside, I think that Rakudo should not restrict the argument to `Positional` as non-`Positional` collection types may implement a `shift` method too. And I don't think that `\a` would be meaningful enough. 

So should I change the parameter to `@a` or not?

Reply:
The problem, again, is unambiguously making something that is really better. "Collection" is implicit in the sigil, for instance. And that's also correct, it's not positional, it's "thing-with-shift". I seem to remember I opened an issue somewhere about that, to create more atomic roles. Anyway, to the point, it's really up to you. I really appreciate the effort for improving the documentation you've put in.

Reply:
I like the idea of more atomic roles. Then, for example, we could document that the "Shiftable" role has a stub

    method shift(Shiftable:D:)

and change the subroutine form to something like

    multi sub shift(Shiftable:D \object)

which will probably give us a more awesome error message when attempting to `shift` something that is not shiftable.

Maybe we could even combine such roles into more complex ones, like:

    role Stacky does Pushable does Poppable {}
    role Queuey does Pushable does Shiftable {}

so we can have a

    class Array does Stacky does Queuey ...

showing what an `Array` can be used for.